sympy/physics/mechanics/_system.py

```diff
@@ -800,18 +800,21 @@ def form_eoms(self, eom_method=KanesMethod):
         if issubclass(eom_method, KanesMethod):
             velocity_constraints = self.holonomic_constraints.diff(
                 dynamicsymbols._t).col_join(self.nonholonomic_constraints)
-            self._eom_method = KanesMethod(
-                self.frame, self.q_ind, self.u_ind, kd_eqs=self.kdes,
-                q_dependent=self.q_dep, u_dependent=self.u_dep,
-                configuration_constraints=self.holonomic_constraints,
-                velocity_constraints=velocity_constraints,
-                forcelist=loads, bodies=self.bodies,
-                explicit_kinematics=False)
+            kwargs = {"frame": self.frame, "q_ind": self.q_ind, "u_ind": self.u_ind,
+                      "kd_eqs": self.kdes, "q_dependent": self.q_dep,
+                      "u_dependent": self.u_dep,
+                      "configuration_constraints": self.holonomic_constraints,
+                      "velocity_constraints": velocity_constraints,
+                      "forcelist": loads, "bodies": self.bodies,
+                      "explicit_kinematics": False, **kwargs}
```

Comment:
What happens if the user passes in one of the kwargs that we are explicitly setting in the function? Will it overwrite? Raise error?

Reply:
It overwrites, as I do also have a test case for

Reply:
Wouldn't that likely fail then? Aren't those things supposed to be managed by this class?

Reply:
The pattern I've typically used for things like this is not to overwrite because if you let someone overwrite things that the class "thought" it had control over, then you basically break a contract with the class. If you don't make the class aware that you are overwriting something it thinks it has control over, then issues often arise.

Reply:
I'm with @moorepants on this one, and having just discussed with @TJStienstra we agree that overwriting should not be allowed. `System` holds a reference to the `KanesMethod`/`LagrangesMethod` instance, so anywhere where these two disagree about a kwarg would be nonsensical. Per the first comment in this thread, an error should be raised.

Reply:
Addressed in 1c4af023695dcaf60f2edfb87350183363e84b42